core/unix/signal_linux_x86.c

```diff
@@ -496,8 +572,35 @@ mcontext_to_sigcontext_simd(sig_full_cxt_t *sc_full, priv_mcontext_t *mc)
                 }
             }
         }
+#ifdef X64
+        if (ZMM_ENABLED()) {
+            kernel_xstate_t *xstate = (kernel_xstate_t *)sc->fpstate;
+            if (sc->fpstate->sw_reserved.magic1 == FP_XSTATE_MAGIC1) {
+                ASSERT(TEST(XCR0_ZMM_HI256, sc->fpstate->sw_reserved.xstate_bv));
```

Comment:
Same 3 comments as in above loop.

Reply:
Same comments / asserts added. Leaving unresolved until 32-bit case agreed on.